File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.py
def solution(number, k):
    N = len(number); number = list(map(int, number))
    
    stack = []
    for i in range(N):
        while stack and stack[-1] < number[i] and k > 0:
            stack.pop()
            k -= 1
        stack.append(number[i])
        # print(stack)
    return "".join(map(str, stack[:len(stack)-k]))

File: 프로그래머스/2/42883. 큰 수 만들기/큰 수 만들기.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

string solution(string number, int k) {
    
    string answer = "";
    vector<char> stack;
    
    for (char ch : number)
    {
        while (!stack.empty() && k > 0 && stack.back() < ch)
        {
            stack.pop_back();
            --k;
        }
        stack.push_back(ch);
    }
    
    while (k > 0)
    {
        stack.pop_back();
        --k;
    }
    
    for (char ch : stack)
    {
        answer += ch;
    }
    
    
    return answer;
}